marklogic-data-hub-central/ui/src/components/entities/matching/compare-values-modal/compare-values-modal.tsx

```diff
@@ -0,0 +1,368 @@
+            allParentKeys.push(localParentKey);
+          } else {
+            if (!allParentKeys.includes(localParentKey)) {
+              allParentKeys.push(property.name);
+            }
+          }
+
+          if (parsedRef.length > 0 && parsedRef[1] === "definitions") {
+            let updatedPropertyPath= propertyPath ? propertyPath : property.name;
+            let URI1Value:any = propertyValueFromPath(updatedPropertyPath, property1);
+            let URI2Value:any = propertyValueFromPath(updatedPropertyPath, property2);
+            let arrLength = 0;
+            if ((URI1Value && Array.isArray(URI1Value)) || (URI2Value && Array.isArray(URI2Value))) {
+              arrLength = URI1Value.length > URI2Value.length ? URI1Value.length : URI2Value.length;
+            }
+            let structuredType = entityDefinitionsArray.find(entity => entity.name === parsedRef[2]);
+            let structuredTypePropertiesArray:any = [];
+            let structuredTypeProperties: any;
+            if (arrLength>0) {
+              let parentKeysTempArray = [...parentKeys];
+              for (let i = 0; i < arrLength; i++) {
+                let allParentKeysTempArray = [...allParentKeys];
+                let structTypeProperties = structuredType?.properties.map((structProperty, structIndex) => {
+                  if (structProperty.datatype === "structured") {
+                    // Recursion to handle nested structured types
+                    counter++;
+                    let parentDefinitionName = structuredType.name;
+                    let immediateParentKey = (parentKey !== "" ? property.name : structProperty.name) + "," + index + counter + i;
+                    let localParentKey = structProperty.name;
+                    let propertyPathUri = propertyPath ? propertyPath : getPropertyPath(parentKeysTempArray, structuredType.name, structProperty.name, undefined, i, property.name);
+                    return parseStructuredProperty(entityDefinitionsArray, structProperty, parentDefinitionName, immediateParentKey, parentKeysTempArray, allParentKeysTempArray, propertyPathUri, i, localParentKey);
+                  } else {
+                    let allParentKeysArray = [...allParentKeysTempArray];
+                    let updatedPropertyPath = getPropertyPath(parentKeysTempArray, structuredType.name, structProperty.name, propertyPath, i, property.name);
+                    let propertyValueInURI1 = propertyValueFromPath(updatedPropertyPath, property1);
+                    let propertyValueInURI2 = propertyValueFromPath(updatedPropertyPath, property2);
+                    let localPropertyPath = getPropertyPathForStructuredProperties(allParentKeysArray, structProperty.name);
+                    let matchedRow = propertyValueInURI1 && propertyValueInURI2 && (JSON.stringify(propertyValueInURI1) === JSON.stringify(propertyValueInURI2)) ? matchedProperties.includes(localPropertyPath) : false;
```

Comment:
Can we test this without JSON.stringify equality comparison ? Just remove the condition "(JSON.stringify(propertyValueInURI1) === JSON.stringify(propertyValueInURI2))" and see if you get the proper results. 
The equality comparison might throw us off in few edge cases where the match type is not Exact.

Reply:
Yes, it returns same results after removing mentioned condition. I will update my PR and will push changes again.